File: 567-permutation-in-string/permutation-in-string.cpp
class Solution {
public:
    bool issame(int freq1[], int freq2[]){
        for(int i = 0; i < 26; i++)
        {
            if(freq1[i] != freq2[i])
            {
                return false;
            }
        }
        return true;
    }
    bool checkInclusion(string s1, string s2) {
        int freq[26] = {0};
        for(int i = 0; i < s1.size(); i++)
        {
            freq[s1[i] - 'a']++;
        }
        int windowsize = s1.size();
        for(int j = 0; j < s2.size(); j++)
        {
            int windowfreq[26] = {0};
            int idx = j;
            int windowidx = 0;
            while(windowidx < windowsize && idx < s2.length())
            {
                windowfreq[s2[idx] - 'a']++;
                idx++;
                windowidx++;
            }
            if(issame(freq, windowfreq))
            {
                return true;
            }
        }
        return false;
    }
};

File: 567-permutation-in-string/permutation-in-string.java
class Solution
{
    private boolean isSame(int[] freq1, int[] freq2)
    {
        for (int i = 0; i < 26; i++)
        {
            if (freq1[i] != freq2[i])
            {
                return false;
            }
        }

        return true;
    }

    public boolean checkInclusion(String s1, String s2)
    {
        int[] freq = new int[26];

        for (int i = 0; i < s1.length(); i++)
        {
            freq[s1.charAt(i) - 'a']++;
        }

        int windowSize = s1.length();

        for (int j = 0; j < s2.length(); j++)
        {
            int[] windowFreq = new int[26];
            int idx = j;
            int windowIdx = 0;

            while (windowIdx < windowSize && idx < s2.length())
            {
                windowFreq[s2.charAt(idx) - 'a']++;
                idx++;
                windowIdx++;
            }

            if (isSame(freq, windowFreq))
            {
                return true;
            }
        }

        return false;
    }
}